### PR TITLE
Block conformance changes on release branches in k/k

### PR DESCRIPTION
**Pull request by nikhita**
Ref: slack discussion - https://kubernetes.slack.com/archives/C78F00H99/p1614202310015700?thread_ts=1613501041.053200&cid=C78F00H99

Follow up to https://github.com/kubernetes/test-infra/pull/21021

cc @hh @dims @justaugustus 
folks on original thread

cc @alejandrox1 @hasheddan @saschagrunert @jeremyrickard 
other sig-release leads

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by nikhita** [comment]
/assign @dims 
for approval

File: config/prow/plugins.yaml [hunk 1]
```diff
@@ -160,6 +160,12 @@ blockades:
   blockregexps:
   - ^pkg/util/[^/]+\.go$
   explanation: "Utility code must be added to a subpackage like pkg/util/flag, not pkg/util directly. See: #49923"
+- repos:
+  - kubernetes/kubernetes
+  blockregexps:
+  - ^test/conformance/testdata/
```
Comment (dims) [inline config/prow/plugins.yaml:166]: @nikhita let's specifically deal with `conformance.yaml`

(as we will run into problems updating OWNERS or deleting bazel if/when we get to it)

Comment (dims) [reply config/prow/plugins.yaml:166]: thanks for taking care of this!

Comment (nikhita) [reply config/prow/plugins.yaml]: Done!

**Review by hasheddan — APPROVED** [review]
/lgtm

thanks @nikhita !

**Comment by spiffxp** [comment]
Reverting this config change for now: https://github.com/kubernetes/test-infra/pull/21092

Will need to fix the NPE and have that deployed before we can make this change again

**Pull request merged**